TLSharp.Tests/TLSharpTests.cs

```diff
@@ -370,12 +369,74 @@ public virtual async Task SendMessageByUserNameTest()
 
             if (user == null)
             {
-                throw new System.Exception("Username was not found: " + UserNameToSendMessage);
+                throw new Exception("Username was not found: " + UserNameToSendMessage);
             }
 
             await client.SendTypingAsync(new TLInputPeerUser() { UserId = user.Id });
             Thread.Sleep(3000);
             await client.SendMessageAsync(new TLInputPeerUser() { UserId = user.Id }, "TEST");
         }
+
+        public virtual async Task GetUpdatesForUser()
+        {
+            IList<TLMessage> newMsgs = new List<TLMessage>();
+            TLUser user = null;
+
+            var client = NewClient();
+            await client.ConnectAsync();
+
+            if (client.IsUserAuthorized())
+                user = client.Session.TLUser;
+
+            else
+            {
+                var hash = await client.SendCodeRequestAsync(NumberToAuthenticate);
+                var code = CodeToAuthenticate; // you can change code in debugger too
+                if (string.IsNullOrWhiteSpace(code))
+                {
+                    throw new Exception("CodeToAuthenticate is empty in the app.config file, fill it with the code you just got now by SMS/Telegram");
+                }
+
+                try
+                {
+                    user = await client.MakeAuthAsync(NumberToAuthenticate, hash, code);
+                }
+                catch (CloudPasswordNeededException)
+                {
+                    var passwordSetting = await client.GetPasswordSetting();
+                    var password = PasswordToAuthenticate;
+                    user = await client.MakeAuthWithPasswordAsync(passwordSetting, password);
+                }
+                catch (InvalidPhoneCodeException ex)
+                {
+                    throw new Exception("CodeToAuthenticate is wrong in the app.config file, fill it with the code you just got now by SMS/Telegram", ex);
+                }
+            }
+
+            client.Updates += (TelegramClient tclient, TLAbsUpdates updates) =>
+            {
+                if (updates is TLUpdates)
+                {
+                    var allupdates = updates as TLUpdates;
+
+                    foreach (var update in allupdates.Updates)
+                    {
+                        if (update is TLUpdateNewMessage)
+                        {
+                            var metaMsg = update as TLUpdateNewMessage;
+                            var msg = metaMsg.Message as TLMessage;
+                            newMsgs.Add(msg);
+                        }
+                    }
+                }
+            };
+
+            await client.MainLoopAsync(new TimeSpan(0, 0, 1));
+
+            // At this point you would send yourself a UPDATE_1 message to trigger update
```

Comment:
let's put this into a Console.WriteLine (and a Debug.WriteLine) so that the person running the test knows he has to do something?